samples/CognitiveServices.TextAnalytics/Generated/CognitiveServicesTextAnalyticsClient.cs

```diff
@@ -26,6 +27,27 @@ protected CognitiveServicesTextAnalyticsClient()
         {
         }
 
+        /// <summary> Initializes a new instance of CognitiveServicesTextAnalyticsClient. </summary>
+        /// <param name="endpoint"> Supported Cognitive Services endpoints (protocol and hostname, for example: https://westus.api.cognitive.microsoft.com). </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public CognitiveServicesTextAnalyticsClient(string endpoint, AzureKeyCredential credential, CognitiveServicesTextAnalyticsClientOptions options = null)
```

Comment:
Why does updating content type add this constructor?

Reply:
This is because of m4 change. The code model adds a new security schema. @timotheeguerin could you comment why this is added?
![image](https://user-images.githubusercontent.com/97225342/162162154-b227f576-92bd-432b-96f3-b99671afa9ba.png)

Reply:
https://github.com/Azure/autorest/blob/main/docs/releases/march-2022.md#modelerfour-security-scheme-generalization

Reply:
@annelo-msft - As per the changes in Modelerfour that @timotheeguerin shared, previously it was restricted with security scheme names and in this case it was looking for the `AzureKey` name and TextAnalytics swagger has [apim_key](https://github.com/Azure/azure-rest-api-specs/blob/c00cf4ad1e2c44ccddb08b98a472cf9822b4d687/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.0/TextAnalytics.json#L14) name. Now it supports ["type": "apiKey"](https://github.com/Azure/azure-rest-api-specs/blob/c00cf4ad1e2c44ccddb08b98a472cf9822b4d687/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.0/TextAnalytics.json#L15). Hence this constructor is generated with `AzureKeyCredential`.

Reply:
@annelo-msft - As per the changes in Modelerfour that @timotheeguerin shared, previously it was restricted with security scheme names(AADToken and AzureKey) and in this case it was looking for the `AzureKey` name and TextAnalytics swagger has [apim_key](https://github.com/Azure/azure-rest-api-specs/blob/c00cf4ad1e2c44ccddb08b98a472cf9822b4d687/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.0/TextAnalytics.json#L14) name. Now it supports ["type": "apiKey"](https://github.com/Azure/azure-rest-api-specs/blob/c00cf4ad1e2c44ccddb08b98a472cf9822b4d687/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.0/TextAnalytics.json#L15), hence this constructor is generated with `AzureKeyCredential`.

Reply:
@annelo-msft looks by design to me.

Reply:
I see, so this is new m4 functionality that was added separately from the contentType changes?

Reply:
Yes, I think so.